src/models/v2/operation-trait.ts

```diff
@@ -54,17 +56,17 @@ export class OperationTrait<J extends v2.OperationTraitObject = v2.OperationTrai
     return externalDocs(this);
   }
 
-  security(): Array<Record<string, { schema: SecuritySchemeInterface; scopes: string[]; }>> {
-    const securitySchemes = this._meta.asyncapi?.parsed?.components?.securitySchemes || {};
+  security(): SecurityRequirements[] {
+    const securitySchemes = this._meta?.asyncapi?.parsed?.components?.securitySchemes || {};
     return (this._json.security || []).map((requirement: any) => {
-      const requirements: Record<string, { schema: SecuritySchemeInterface; scopes: string[]; }> = {};
+      const requirements: SecurityRequirement[] = [];
       Object.entries(requirement).forEach(([security, scopes]) => {
-        requirements[security] = {
-          schema: this.createModel(SecurityScheme, securitySchemes[security], { id: security, pointer: `/components/securitySchemes/${security}` }),
-          scopes: scopes as Array<string>,
-        }
+        const scheme = this.createModel(SecurityScheme, securitySchemes[security], { id: security, pointer: `/components/securitySchemes/${security}` });
+        requirements.push(
+          this.createModel(SecurityRequirement, scopes, { id: security, scheme: scheme, pointer: `` }) // TODO pointer??
```

Comment:
Pointer for `SecurityRequirement` should be as (example):

```
/channels/{channel}/{operation}/security/{index of requirements}/{security variable - name of security schemes}
```

Why? Because we point to a single scheme of a single security requirement :) I hope this is clear. Let me know of any misunderstandings!